### PR TITLE
VLE/QRW-54/QRW-58/reg_sys

### PR DESCRIPTION
**Pull request by Akaud**
Created registration and login page, added possibility to login using Google Account

### CODE REVIEW

File: reg_sys/templates/reg_sys/profile.html [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{% extends 'reg_sys/base.html'%}
+{% block content %}
+    <h1>Hello {{ user }}</h1>
+{% endblock %}
```
Comment (Strexas) [inline reg_sys/templates/reg_sys/profile.html:4]: profile template can't be in registration system

File: reg_sys/urls.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+"""urls.py"""
+from django.urls import path
+from .import views
```
Comment (Strexas) [inline reg_sys/urls.py:3]: space after .

File: reg_sys/urls.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+"""urls.py"""
```
Comment (Strexas) [inline reg_sys/urls.py:1]: Use better documentation, what is the purpose of this file. You may refer to qr_web/urls.py

File: reg_sys/views.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
+"""views.py"""
```
Comment (Strexas) [inline reg_sys/views.py:1]: same as before

File: reg_sys/views.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+from django.shortcuts import render, redirect
+from django.contrib.auth import authenticate
+from django.contrib import messages, auth
+from django.contrib.auth.decorators import login_required
+from .forms import UserRegisterForm, LoginForm
+# Create your views here.
```
Comment (Strexas) [inline reg_sys/views.py:7]: purpose of this line?

File: reg_sys/views.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+
+def home(request):
+    """Homepage"""
+    return render(request, 'reg_sys/startpage.html')
+
+
+# --Register a user
```
Comment (Strexas) [inline reg_sys/views.py:15]: Description of function should be in documentation of function

File: reg_sys/views.py [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+            messages.success(request, f'Account created for {form.cleaned_data["username"]}')
+            return redirect('login')
+    else:
+        form = UserRegisterForm()
+    context = {'form': form}
+    return render(request, 'reg_sys/register.html', context=context)
+
+
+# - Login a user
+def user_login(request):
```
Comment (Strexas) [inline reg_sys/views.py:31]: annotate parameters of function

File: reg_sys/views.py [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+            user = authenticate(request, username=username, password=password)
+            messages.success(request, f'{form.cleaned_data["username"]} successfully logged in')
+            if user is not None:
+                auth.login(request, user)
+                return redirect('profile')
+    context = {'form': form}
+    return render(request, 'reg_sys/login.html', context=context)
+
+
+# - User logout
```
Comment (Strexas) [inline reg_sys/views.py:48]: same as before

File: reg_sys/views.py [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+def user_logout(request):
+    """Logout process"""
+    auth.logout(request)
+    return redirect('login')
+
+
+# - Profile page
```
Comment (Strexas) [inline reg_sys/views.py:55]: same as before

File: static/css/styles.css [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+.button {
+    width: 200px;
+    transition: 0.7s;
+}
+
+.button:hover {
+    width: 300px;
+}
```
Comment (Strexas) [inline static/css/styles.css:21]: css in Nojus' job, from your side html only is required

File: reg_sys/urls.py [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+from django.urls import path
+from .import views
+
+urlpatterns = [
```
Comment (Strexas) [inline reg_sys/urls.py:5]: naming of variable is not snake_case

**Review by Strexas — CHANGES_REQUESTED** [review]
Changes are requested + resolve merging conflicts

**Comment by Strexas** [comment]
Change name from `reg_sys` to move convinient reg is not official short

File: reg_sys/models.py [hunk 12]
```diff
@@ -0,0 +1,2 @@
+"""Create your models here."""
+# from django.db import models
```
[the remark this answers is not in the document]
Comment (Akaud) [reply reg_sys/models.py:2]: image

**Pull request closed without merging**